src/ring_theory/derivation.lean

```diff
@@ -0,0 +1,223 @@
+/-
+Copyright © 2020 Nicolò Cavalleri. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Nicolò Cavalleri.
+-/
+
+import algebra.lie_algebra
+import tactic
+
+/-!
+# Derivations
+
+This file defines derivation. A derivation from the `R`-agebra `A` to the `A`-module `M` is an `R`
+linear map that satisfy the Leibniz rule.
+
+## Notation
+
+The notation `⁅D1, D2⁆` is used for the commutator of two derivations.
+
+TODO: this file is just a stub to go on with some PRs in the geometry section. It only
+implements the definition of derivations in commutative algebra. This will soon change: as soon
+as bimodules will be there in mathlib I will change this file to take into account the
+non-commutative case. Any development on the theory of derivations is discouraged until the
+definitive definition of derivation will be implemented.
+
+Also note that this file implements bundled derivations only by now. To be better integrated with
+the algebra section of mathlib it would probably be better to define a notion of `is_derivation` and
+link it to bundled derivations.
+-/
+
+open algebra ring_hom
+
+instance algebra.to_is_scalar_tower  {R : Type*} {A : Type*} [comm_semiring R] [semiring A]
+  [algebra R A] : is_scalar_tower R A A :=
+⟨λ r a b, by simp only [smul_def]; exact mul_smul _ _ _⟩
```

Comment:
I believe this is the same as [`is_algebra_tower.right`](https://github.com/leanprover-community/mathlib/blob/master/src/ring_theory/algebra_tower.lean#L98). Can you use that instance here (after generalizing it as appropriate)?

Reply:
Well I did try to remove all instances to see if Lean could find other instances, but it did not find this one. Could it be sensible to move just that instance to algebra.algebra? I mean so that one needs not go looking for where it is, since I'd expect that having algebras and is_scalar_tower should be enough to expect the instance to be there. Also I wonder why keeping the name `is_algebra_tower`? I think things would be way clearer if only `is_scalar_tower` were there